db/index_changes.go

```diff
@@ -402,9 +403,15 @@ func (db *Database) vectorChangesFeed(channel string, options ChangesOptions) (<
 		if options.Since.TriggeredByClock != nil {
 			for i := 0; i < len(log); i++ {
 				logEntry := log[i]
-				// If sequence is less than the backfillTo clock sequence for its vbucket, send as backfill (i.e. with triggered by)
-				isBackfill := logEntry.Sequence <= options.Since.TriggeredByClock.GetSequence(logEntry.VbNo)
 
+				// If vb.seq for the entry is earlier than the vb.seq that triggered this channel's backfill, send as backfill.
+				isBackfill := false
+				if base.CompareVbAndSequence(logEntry.VbNo, logEntry.Sequence, options.Since.TriggeredByVbNo, options.Since.TriggeredBy) == -1 {
+					isBackfill = true
+				}
+				if logEntry.VbNo < options.Since.TriggeredByVbNo || (logEntry.VbNo == options.Since.TriggeredByVbNo && logEntry.Sequence < options.Since.TriggeredBy) {
```

Comment:
Sheesh - thanks for catching.  That's what I get for trying to push up the fix in between meetings.